using-timescaledb/custom-jobs.md

```diff
@@ -0,0 +1,221 @@
+# Custom Jobs
+
+Custom jobs allow you to run functions and procedures implemented in a
+language of your choosing on a schedule within TimescaleDB. This allows
+automatic periodic tasks that are not covered by existing policies and
+even enhancing existing policies with additional functionality.
+
+### Creating a custom job procedure [](create-procedure)
+
+The signature for custom jobs is (job_id INT, config JSONB).
+The content of the config JSONB is completely up to the job and may
+also be NULL if no variables are required.
+
+Template for a custom job procedure.
+
+```sql
+CREATE OR REPLACE PROCEDURE custom_job(job_id int, config jsonb) LANGUAGE PLPGSQL AS
+$$
+BEGIN
+	RAISE NOTICE 'Executing job % with config %', job_id, config;
+END
+$$;
+```
+
+### Registering your custom job [](register)
+
+When registering the custom job `job_id` and `config` are stored in the
+TimescaleDB catalog. The `config` JSONB can be modified with `alter_job`.
+`job_id` and `config` will be passed as arguments when the procedure is
+executed as background process or when run with `run_job`.
+
+```sql
+SELECT add_job('custom_job','1h');
+```
+
+Register the created job with the automation framework. `add_job` returns the job_id
+which can be used to execute the job manually with `run_job`.
+
+```sql
+CALL run_job(1000);
+```
+
+Execute job with id 1000 in current session.
+
+### Testing/Debugging jobs [](testing)
+
+Any background worker job can be run in foreground when executed with `run_job`. This can
+be useful to debug problems when combined with increased log level.
+
+### Altering / Dropping custom jobs [](alter-delete)
+
+You can alter the config or scheduling parameters with `alter_job`.
+
+```sql
+SELECT alter_job(1000, config => '{"hypertable":"metrics"}');
+```
+
+Replaces the config for job with id 1000 with the specified JSON.
```

Comment:
Please clarify:  Does this replace the _entire_ JSON, or can it be used to replace individuals _keys_ within the JSON?  My guess is the former, but useful to clarify.